File: C++/A/48A-rockpaperscissors.cpp
#include <bits/stdc++.h>
using namespace std;

char determine_winner(string a, string b, string c)
{
    if ((a == "rock" && b == "scissors" && c == "scissors") || (a == "scissors" && b == "paper" && c == "paper") || (a == "paper" && b == "rock" && c == "rock"))
        return 'F';
    else if ((b == "rock" && a == "scissors" && c == "scissors") || (b == "scissors" && a == "paper" && c == "paper") || (b == "paper" && a == "rock" && c == "rock"))
        return 'M';
    else if ((c == "rock" && b == "scissors" && a == "scissors") || (c == "scissors" && b == "paper" && a == "paper") || (c == "paper" && b == "rock" && a == "rock"))
        return 'S';
    else
        return '?';
}

int main()
{
    string fyodor, matroskin, sharic;
    cin >> fyodor >> matroskin >> sharic;
    cout << determine_winner(fyodor, matroskin, sharic);
}

File: Python/A/48A-rockpaperscissors.py
def determine_winner(fyodor, matroskin, sharic):
    if (fyodor == "rock" and matroskin == "scissors" and sharic == "scissors") or (fyodor == "scissors" and matroskin == "paper" and sharic == "paper") or (fyodor == "paper" and matroskin == "rock" and sharic == "rock"):
        return "F"
    elif (matroskin == "rock" and fyodor == "scissors" and sharic == "scissors") or (matroskin == "scissors" and fyodor == "paper" and sharic == "paper") or (matroskin == "paper" and fyodor == "rock" and sharic == "rock"):
        return "M"
    elif (sharic == "rock" and matroskin == "scissors" and fyodor == "scissors") or (sharic == "scissors" and matroskin == "paper" and fyodor == "paper") or (sharic == "paper" and matroskin == "rock" and fyodor == "rock"):
        return "S"
    else:
        return "?"


fyodor = input()
matroskin = input()
sharic = input()

winner = determine_winner(fyodor, matroskin, sharic)
print(winner)
